### PR TITLE
fix: some scam bugs

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-2739 #MF-2740 #MF-2741

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const sns: Plugin.SNSAdaptor.Definition = {
         const links = useMemo(() => {
             const x = extractTextFromTypedMessage(props.message)
             if (x.none) return null
-            return parseURLs(x.val)
+            return parseURLs(x.val, false)
```
Comment (Jack-Works) [inline packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx:36]: why you're going to use an invalid URL?

File: packages/plugins/ScamWarning/src/locales/en-US.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
     "__plugin_name": "Safety Warning",
     "__plugin_description": "Detecting whether user-published crypto websites are risky or malicious. Help to save your digital assets.",
-    "warning_description": "This domain is currently on the Mask Network domain warning list."
+    "warning_description": "These domain are currently on the Mask Network domain warning list which are a source of malware, phishing and spam."
```
Comment (guanbinrui) [inline packages/plugins/ScamWarning/src/locales/en-US.json]: These domains...

File: packages/web3-providers/src/cryptoscam-db/index.ts [hunk 3]
```diff
@@ -16,17 +16,36 @@ export class CryptoScamDBAPI implements ScamWarningAPI.Provider {
...
+            const url = new URL(link)
 
-        return fetchJSON<ScamWarningAPI.Info>(urlcat(baseURL, `${key}.json`))
+            if (!filter.has(url.host)) return
+
+            const result = await fetchJSON<ScamWarningAPI.Info>(urlcat(baseURL, `${url.host}.json`))
+
+            if (!result) return
+
+            if (url.pathname.toLowerCase() === new URL(result.url).pathname.toLowerCase()) {
```
Comment (guanbinrui) [inline packages/web3-providers/src/cryptoscam-db/index.ts:34]: Why we need to concern about equity of these two pathnames?

File: packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx (hunk 1 above)
Comment (Lanttcat) [reply packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx:36]: I hope get the `a.com` from message which not include protocol.

File: packages/web3-providers/src/cryptoscam-db/index.ts (hunk 3 above)
Comment (Lanttcat) [reply packages/web3-providers/src/cryptoscam-db/index.ts:34]: The cryptoscamdb have some data like:
```json
{
            "id": "***",
            "name": "docs.google.com",
            "url": "https://docs.google.com/forms/d/e/1FAIpQLSeRw1qALR7-rbvOKRM-Wgs1TxmAN5BC7_XzATFiHIVRuRlegQ/viewform",
}

```

We should not render plugin for `https://docs.google.com/some-other-safe-link`

Comment (guanbinrui) [reply packages/web3-providers/src/cryptoscam-db/index.ts:34]: What does an empty pathname like in CryptoScamDB? Here is a case that will fail to detect successfully:

+ Given Link: https://scam-link.com/path-to-some-thing
+ ScamDB: `name: scam-link.com` and `url: https://scam-link.com`

We know that the given one is a scam link, but ScamDB thinks all paths under `https://scam-link.com` are unsafe.

If there is no pathname found in `result.url`, we don't need to compare pathnames anymore.

File: packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx (hunk 1 above)
Comment (guanbinrui) [reply packages/plugins/ScamWarning/src/SNSAdaptor/index.tsx:36]: Maybe you could create another one: `parseLinks`.

**Pull request merged**